compiler/rustc_mir_build/src/build/coverageinfo.rs

```diff
@@ -385,4 +409,25 @@ impl Builder<'_, '_> {
             mcdc_state.record_conditions(logical_op, span);
         }
     }
+
+    pub(crate) fn mcdc_increment_depth_if_enabled(&mut self) {
+        if let Some(branch_info) = self.coverage_branch_info.as_mut()
+            && let Some(mcdc_state) = branch_info.mcdc_state.as_mut()
+        {
+            mcdc_state.decision_ctx_stack.push(MCDCDecisionCtx::default());
+        };
+    }
+
+    pub(crate) fn mcdc_decrement_depth_if_enabled(&mut self) {
+        if let Some(branch_info) = self.coverage_branch_info.as_mut()
+            && let Some(mcdc_state) = branch_info.mcdc_state.as_mut()
+        {
+            #[allow(unused)]
+            let Some(MCDCDecisionCtx { decision_stack, processing_decision }) =
+                mcdc_state.decision_ctx_stack.pop()
+            else {
+                bug!("Unexpected empty Decision stack");
+            };
```

Comment:
Is it intentional? Could we guard it with `mcdc_state.decision_ctx_stack.pop().is_none()` and remove `unused`?

Reply:
thanks for pointing this out ! I didn't notice it while rebasing.